server/src/Application/Services/Flashcard/FlashcardFileImporter.cs

```diff
@@ -0,0 +1,63 @@
+using System.Threading;
+using StudyZen.Application.Dtos;
+
+
+namespace StudyZen.Application.Services
+{
+    public class FlashcardFileImporter
+    {
+        private readonly IFlashcardService _flashcardService;
+
+        public FlashcardFileImporter(IFlashcardService flashcardService)
+        {
+            _flashcardService = flashcardService;
+        }
+
+       public void ImportFlashcardsFromCsvStream(Stream stream)
+        {
+            try
+            {
+                using (var reader = new StreamReader(stream))
+                {
+                    var flashcardImportThreads = new List<Thread>();
+
+                    while (!reader.EndOfStream)
+                    {
+                        var line = reader.ReadLine();
+                        if (string.IsNullOrWhiteSpace(line))
+                        {
+                            continue;
+                        }
+
+                        var values = line.Split(',');
+
+                        if (values.Length == 3 && int.TryParse(values[0], out int flashcardSetId))
+                        {
+                            string question = values[1];
+                            string answer = values[2];
+
+                            var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, question, answer);
+
+                            var thread = new Thread(() =>
+                            {
+                                _flashcardService.CreateFlashcard(createFlashcardDto);
```

Comment:
Flashcard service (and generic repository) is not concurrency safe, so this code will break.

In addition, this code is responsible for multiple things: reading and writing. It should be separated. This would also solve concurrency problems.

We could have a method which reads flashcards from file (look into Parallel.ForEach: https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/how-to-write-a-simple-parallel-foreach-loop) and add a method to FlashcardService which accepts a list of flashcards to create.

Then, if this class just reads the flashcards from file, it could be put under Infrastructure project.